srcmedia/ts/members-map.ts

```diff
@@ -0,0 +1,127 @@
+import { map, icon, marker, tileLayer, control, featureGroup, PopupEvent } from 'leaflet'
+import { TiledMapLayer } from 'esri-leaflet'
+
+// map data - defined in the global scope outside this module, in the template
+type Address = {
+    name: string,
+    street_address: string,
+    city: string,
+    postal_code: number,
+    latitude: number,
+    longitude: number,
+    start_date: string,
+    end_date: string,
+}
+
+declare const address_data: Array<Address>
+
+/*
+ * map object with custom zoom control in bottom right
+ */
+const target = document.getElementById('address-map') as HTMLDivElement
+const addressMap = map(target, { zoomControl: false })
+const zoomControl = control.zoom({ position: 'bottomright' })
+
+zoomControl.addTo(addressMap)
+
+/*
+* basic tiled basemap from mapbox
+*/
+const mapboxToken = 'pk.eyJ1IjoicHJpbmNldG9uLWNkaCIsImEiOiJjazJrd2lybnEwMHdsM2JvM3UyMHUwbm02In0.4GJpwErZHkJH1DU-E-72OA'
+const basemap = tileLayer(`https://api.mapbox.com/styles/v1/mapbox/light-v10/tiles/{z}/{x}/{y}?access_token=${mapboxToken}`, {
```

Comment:
Possibly the basemap should also be in a config, although that seems less important to me than some of the other items.